File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/cs/AltovaText/TableLike/FLF/FLFFormat.cs
[call GenerateFileHeader("FLFFormat.cs")]
namespace Altova.TextParser.TableLike.FLF
{
	/// <summary>
	/// Encapsulates the specification of a fixed length fields format.
	/// </summary>
	public class FLFFormat
	{
		#region Implementation Detail:
		bool mAssumeRecordDelimitersPresent = false;
		char mFillCharacter = ' ';
		char\[\] mRecordDelimiters = new char\[\]
			{
				'\\n',
				'\\r'
			};
		#endregion
		/// <summary>
		/// Get/sets whether record delimiters are used.
		/// </summary>
		public bool AssumeRecordDelimitersPresent
		{
			get
			{
				return mAssumeRecordDelimitersPresent;
			}
			set
			{
				mAssumeRecordDelimitersPresent = value;
			}
		}
		/// <summary>
		/// Get/sets the character used to fill fields up to their specified length.
		/// </summary>
		public char FillCharacter
		{
			get
			{
				return mFillCharacter;
			}
			set
			{
				mFillCharacter = value;
			}
		}
		/// <summary>
		/// Checks whether a given character is a record delimiter (\\n or \\r).
		/// </summary>
		/// <param name="rhs">the character to check</param>
		/// <returns>true if the character is a record delimiter, otherwise false</returns>
		public bool IsRecordDelimiter(char rhs)
		{
			foreach (char c in mRecordDelimiters)
				if (c == rhs) return true;
			return false;
		}
	}
}

File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/cpp/AltovaText/FLFFormat.cpp
[call GenerateFileHeader("FLFFormat.cpp")]
#include "stdafx.h"
#include "FLFFormat.h"

namespace altova
{
namespace text
{
namespace tablelike
{
namespace flf
{

CFormat::CFormat()
:	m_AssumeRecordDelimiters(false)
,	m_FillCharacter(_T(' '))
{
	m_RecordDelimiters\[0\]= _T('\\r');
	m_RecordDelimiters\[1\]= _T('\\n'); 
}

bool CFormat::GetAssumeRecordDelimiters() const
{
	return m_AssumeRecordDelimiters;
}

void CFormat::SetAssumeRecordDelimiters(bool rhs)
{
	m_AssumeRecordDelimiters= rhs;
}

TCHAR CFormat::GetFillCharacter() const
{
	return m_FillCharacter;
}

void CFormat::SetFillCharacter(TCHAR rhs)
{
	m_FillCharacter= rhs;
}

bool CFormat::IsRecordDelimiter(TCHAR rhs) const
{
	for (size_t i= 0; i<2; ++i)
	{
		if (rhs == m_RecordDelimiters\[i\])
			return true;
	}
	return false;
}

} // namespace flf
} // namespace tablelike
} // namespace text
} // namespace altova

